stdlib/Distributed/src/managers.jl

```diff
@@ -145,7 +145,34 @@ end
 show(io::IO, manager::SSHManager) = println(io, "SSHManager(machines=", manager.machines, ")")
 
 
-function launch_on_machine(manager::SSHManager, machine, cnt, params, launched, launch_ntfy::Condition)
+function parse_machine(machine::AbstractString)
+    hoststr = ""
+    portstr = ""
+
+    if machine[begin] == '['  # ipv6 bracket notation (RFC 2732)
+        ipv6_end = findlast(']', machine)
+        hoststr = machine[begin+1 : ipv6_end - 1]
+        machine_def = split(machine[ipv6_end : end] , ':')
+    else    # ipv4
+        machine_def = split(machine, ':')
+        hoststr = machine_def[1]
+    end
+
+    if length(machine_def) > 2
+        throw(ArgumentError("invalid machine definition format string: invalid port format \"$machine_def\""))
+    end
+
+    if length(machine_def) == 2
+        portstr = machine_def[2]
+        if !all(isdigit, portstr) || (p = parse(Int,portstr); p < 1 || p > 65535)
+            msg = "invalid machine definition format string: invalid port format \"$machine_def\""
+            throw(ArgumentError(msg))
+        end
+    end
+    (hoststr, portstr)
```

Comment:
Return the actual port number here or `nothing` if there is no port number supplied.

Reply:
changed to return the actual parsed port number or nothing and tests updated in aa7a0b6